src/s3path.jl

```diff
@@ -333,14 +333,59 @@ function FilePathsBase.sync(f::Function, src::AbstractPath, dst::S3Path; delete=
     end
 end
 
+
+# for some reason, sometimes we get back a `Pair`
+# other times a `AbstractDict`.
+function _pair_or_dict_get(p::Pair, k)
+    first(p) == k || return nothing
+    return last(p)
+end
+_pair_or_dict_get(d::AbstractDict, k) = get(d, k, nothing)
+
+function _retrieve_prefixes!(results, objects, prefix_key, chop_head)
+    objects === nothing && return nothing
+
+    rm_key = s -> chop(s, head=chop_head, tail=0)
+
+    for p in objects
+        prefix = _pair_or_dict_get(p, prefix_key)
+        
+        if prefix !== nothing
+            push!(results, rm_key(prefix))
+        end
+    end
+    
+    return nothing
+end
+
+function _readdir_add_results!(results, response, key_length)
+    sizehint!(results, length(results) + parse(Int, response["KeyCount"]))
+
+    _retrieve_prefixes!(results, get(response, "CommonPrefixes", nothing), "Prefix", key_length)
+    _retrieve_prefixes!(results, get(response, "Contents", nothing), "Key", key_length)
+
+    return get(r, "NextContinuationToken", nothing)
+end
```

Comment:
Can't leave a comment on the line above :/

We're trying to `get(r)` but it DNE?

Reply:
ohh whoops, that should be `response` as well, sorry. I'll push a fix